src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1484,13 +1484,9 @@ void TorrentHandle::toggleFirstLastPiecePriority()
 void TorrentHandle::pause()
 {
     if (m_startupState != Started) return;
-    if (m_pauseWhenReady) return;
-    if (isChecking()) {
-        m_pauseWhenReady = true;
-        return;
-    }
-
     if (isPaused()) return;
+    if (isChecking())
+        m_pauseWhenReady = false;
```

Comment:
Well, I've just done some code review. This flag is cleared in `handleTorrentPausedAlert()` so it seems you don't really need to clear it here.

Reply:
Reverted.